keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ArtifactRepository.kt

```diff
@@ -24,7 +24,7 @@ interface ArtifactRepository : PeriodicallyCheckedRepository<DeliveryArtifact> {
 
   fun get(name: String, type: ArtifactType, reference: String, deliveryConfigName: String): DeliveryArtifact
 
-  fun get(deliveryConfigName: String, reference: String, type: ArtifactType): DeliveryArtifact
+  fun get(deliveryConfigName: String, reference: String): DeliveryArtifact
```

Comment:
I think this is a good move. The only concern I have is that currently `DeliveryArtifact.reference` defaults to `DeliveryArtifact.name`, which, to your point in the PR description, is only unique when combined with the artifact type. I feel like we should add an additional check in the delivery config validation to ensure artifact references are unique. Maybe in a separate PR.

Reply:
There is already a check validation to make sure references are unique, so we're all set here. Defaulting to name is okay because we will fail the validation if you have two artifacts with the same name and you don't define a different reference.